lib/domain/dtos/EndRunDto.js

```diff
@@ -14,8 +14,15 @@
 const Joi = require('joi');
 
 const BodyDto = Joi.object({
-    timeO2End: Joi.date(),
-    timeTrgEnd: Joi.date(),
+    timeO2Start: Joi.date().allow(-1).optional(),
+    timeTrgStart: Joi.date().allow(-1).optional(),
+    timeO2End: Joi.date().allow(-1).optional(),
+    timeTrgEnd: Joi.date().allow(-1).optional(),
```

Comment:
Can we think about another solution than -1? Sending nullable timestamp for example. 
Because here the -1 seems to come out of nowhere when you do not know the context, and actually negative timestamps may have a meaning (probably not in this application for sure)
I really think data must be self explicit, and using a not null value to represent null is really error prone

Reply:
The go API can only send int64 types so we are restricted to numbers. Furthermore, we do not want to save dates before 2020 so that's why the decision was made to use -1 as a null. For the users that use the API the values still need to accept non-existing/empty values so that's why there is still an optional option with the validation. Small example [here](https://go.dev/play/p/w8Qdhi1fTMX)

Reply:
Discussed this with george as well we are going to change it to be handled within the `BookkeepingApi.go`